### DISCUSSION TITLE
Telnet on ESP32 port

### DISCUSSION THREAD

**Discussion by JayPalm** [micropython/micropython#10411] [Libraries & Drivers]
I see there is a [telnet module](https://github.com/micropython/micropython/blob/e3291e1801f27e49ac0e6be275c742777507207f/ports/cc3200/telnet/telnet.c) in the cc3200 port. Is there a reason this isn't in other ports? Would it work in the ESP32 port? Has anyone tried just adding the files/cmake config and building the firmware?

**Comment by robert-hh** [comment]
>  Would it work in the ESP32 port? 

No
There is a Python implementation of Telnet here: https://github.com/cpopp/MicroTelnetServer
It works, using similar basic mechanisms as Webrepl and the Python ftp server.

**Reply by rkompass** [reply]
Would this be something for MP lib?

**Comment by Josverl** [comment]
Micro telnet is a bit limited a, mostly as it does not support AuthN making it dangerous outside of cursirary use IMO. I'm sure there is a fork with at least PWD but no TLS as far as I am aware.
